go-controller/pkg/node/gateway.go

```diff
@@ -172,67 +173,36 @@ func (g *gateway) Run(stopChan <-chan struct{}, wg *sync.WaitGroup) {
 	}
 }
 
-func gatewayInitInternal(nodeName, gwIntf string, subnets []*net.IPNet, gwNextHops []net.IP, nodeAnnotator kube.Annotator) (
-	string, string, net.HardwareAddr, []*net.IPNet, error) {
+func gatewayInitInternal(nodeName, gwIntf, egressGatewayIntf string, subnets []*net.IPNet, gwNextHops []net.IP, nodeAnnotator kube.Annotator) (
+	*bridgeConfiguration, *bridgeConfiguration, error) {
 
-	var bridgeName string
-	var uplinkName string
-	var brCreated bool
-	var err error
-
-	if bridgeName, _, err = util.RunOVSVsctl("--", "port-to-br", gwIntf); err == nil {
-		// This is an OVS bridge's internal port
-		uplinkName, err = util.GetNicName(bridgeName)
-		if err != nil {
-			return bridgeName, uplinkName, nil, nil, err
-		}
-	} else if _, _, err := util.RunOVSVsctl("--", "br-exists", gwIntf); err != nil {
-		// This is not a OVS bridge. We need to create a OVS bridge
-		// and add cluster.GatewayIntf as a port of that bridge.
-		bridgeName, err = util.NicToBridge(gwIntf)
-		if err != nil {
-			return bridgeName, uplinkName, nil, nil, fmt.Errorf("failed to convert %s to OVS bridge: %v", gwIntf, err)
-		}
-		uplinkName = gwIntf
-		gwIntf = bridgeName
-		brCreated = true
-	} else {
-		// gateway interface is an OVS bridge
-		uplinkName, err = getIntfName(gwIntf)
-		if err != nil {
-			return bridgeName, uplinkName, nil, nil, err
-		}
-		bridgeName = gwIntf
-	}
-
-	// Now, we get IP addresses from OVS bridge. If IP does not exist,
-	// error out.
-	ips, err := getNetworkInterfaceIPAddresses(gwIntf)
+	gatewayBridge, err := bridgeForInterface(gwIntf, nodeName, types.PhysicalNetworkName)
 	if err != nil {
-		return bridgeName, uplinkName, nil, nil, fmt.Errorf("failed to get interface details for %s (%v)",
-			gwIntf, err)
+		return nil, nil, errors.Wrapf(err, "Bridge for interface failed for %s", gwIntf)
 	}
-	ifaceID, macAddress, err := bridgedGatewayNodeSetup(nodeName, bridgeName, gwIntf,
-		types.PhysicalNetworkName, brCreated)
-	if err != nil {
-		return bridgeName, uplinkName, nil, nil, fmt.Errorf("failed to set up shared interface gateway: %v", err)
+	egressGWBridge := &bridgeConfiguration{}
```

Comment:
as we discussed on slack, you can just initialize this to a copy of gatewayBridge, then remove the need to pass the param to ovnkube-master

Reply:
I ended up leaving this empty and checking for the presence of the interface in ovnkube-master